ledger/participant-integration-api/src/test/suite/scala/platform/index/BuffersUpdaterSpec.scala

```diff
@@ -0,0 +1,302 @@
+
+      eventually {
+        shutdownCodeCapture.get() shouldBe 1
+      }
+    }
+  }
+
+  "convertToContractStateEvents" should {
+    "convert TransactionLogUpdate.LedgerEndMarker to ContractStateEvent.LedgerEndMarker" in {
+      BuffersUpdater
+        .convertToContractStateEvents(
+          TransactionLogUpdate.LedgerEndMarker(
+            eventOffset = Offset.fromByteArray(BigInt(1337L).toByteArray),
+            eventSequentialId = 9876L,
+          )
+        )
+        .toVector shouldBe Vector(
+        ContractStateEvent.LedgerEndMarker(
+          eventOffset = Offset.fromByteArray(BigInt(1337L).toByteArray),
+          eventSequentialId = 9876L,
+        )
+      )
+    }
+
+    "convert TransactionLogUpdate.Transaction to a series of ContractStateEvent (created/archived)" in {
+      val createdCid = ContractId.assertFromString("#createdCid")
+      val createdOffset = Offset.fromByteArray(BigInt(1337L).toByteArray)
+      val createdEventSeqId = 9876L
+      val createdLedgerEffectiveTime = Instant.ofEpochMilli(987654321L)
+      val createdTemplateId = Ref.Identifier.assertFromString("create:template:id")
+      val createdContractKey =
+        VersionedValue[ContractId](TransactionVersion.VDev, ValueInt64(1337L))
+      val createdFlatEventWitnesses = Set("alice", "charlie")
+      val createArgument = VersionedValue(TransactionVersion.VDev, ValueText("arg"))
+      val createAgreement = "agreement"
+
+      val exercisedCid = ContractId.assertFromString("#exercisedCid")
+      val exercisedKey = VersionedValue[ContractId](TransactionVersion.VDev, ValueInt64(8974L))
+      val exercisedTemplateId = Ref.Identifier.assertFromString("exercised:template:id")
+      val exercisedFlatEventWitnesses = Set("bob", "dan")
+      val exercisedOffset = Offset.fromByteArray(BigInt(1337L).toByteArray)
+      val exercisedEventSequentialId = 9876L
+
+      val consumingExercise = TransactionLogUpdate.ExercisedEvent(
+        contractId = exercisedCid,
+        contractKey = Some(exercisedKey),
+        templateId = exercisedTemplateId,
+        flatEventWitnesses = exercisedFlatEventWitnesses,
+        eventOffset = exercisedOffset,
+        eventSequentialId = exercisedEventSequentialId,
+        consuming = true,
+        transactionId = null,
+        nodeIndex = 0,
+        eventId = null,
+        ledgerEffectiveTime = null,
+        commandId = null,
+        workflowId = null,
+        treeEventWitnesses = null,
+        choice = null,
+        actingParties = null,
+        children = null,
+        exerciseArgument = null,
+        exerciseResult = null,
+      )
+      val transaction = TransactionLogUpdate.Transaction(
+        transactionId = "some-tx-id",
+        commandId = "some-cmd-id",
+        workflowId = "some-workflow-id",
+        effectiveAt = Instant.EPOCH,
+        offset = Offset.beforeBegin,
+        events = Vector(
+          TransactionLogUpdate.CreatedEvent(
```

Comment:
Don't you want to factor it out to a val to increase readability, same as in consumingExercise?

Reply:
Done